File: 383-ransom-note/383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string r, string m) {
        map<char,int>mp;
        for(auto i:m) mp[i]++;
        
        for(auto i:r){
            if(mp[i]) mp[i]--;
            else return false;
        }
        return true;
    }
};

File: 383-ransom-note/383-ransom-note.py
class Solution:
    def canConstruct(self, r: str, m: str) -> bool:
        dp=defaultdict(int)
        for i in m:
            dp[i]+=1
        dp2=defaultdict(int)
        
        for i in r:
            dp2[i]+=1
            
        for i in r:
            if dp[i]<dp2[i]:
                return False
        return True